server/api/utils/clients/iguazio.py

```diff
@@ -752,11 +752,22 @@ def _transform_mlrun_project_to_iguazio_project(
             body["data"]["attributes"]["owner_username"] = project.spec.owner
 
         if project.spec.default_function_node_selector is not None:
-            body["data"]["attributes"]["default_function_node_selector"] = (
-                Client._transform_mlrun_labels_to_iguazio_labels(
-                    project.spec.default_function_node_selector
+            # This feature requires support for project-level default_function_node_selector,
+            # which is available starting from version 3.5.5.
+            # We are adding this validation to maintain backward compatibility with older versions of Iguazio.
+            if mlrun.utils.helpers.is_igz_version_sufficient("3.5.5"):
+                body["data"]["attributes"]["default_function_node_selector"] = (
+                    Client._transform_mlrun_labels_to_iguazio_labels(
+                        project.spec.default_function_node_selector
+                    )
                 )
-            )
+            else:
+                logger.warning(
+                    "User attempted to use a feature requiring project-level default_function_node_selector,"
+                    " but the installed version of Iguazio is not compatible. "
+                    "Project created without the field in Iguazio."
+                )
```

Comment:
Warning is a bit much for my taste, suggest giving less worrying log and log some informative parameters
```suggestion
                logger.debug(
                    "Omitting project-level default function node selector from Iguazio project as Iguazio version is insufficient", 
                    igz_version=mlconf.igz_version,
                    project_name= project.metadata.name
                )
```